echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy

```diff
@@ -16,79 +16,113 @@
+  long syncInterval
 
-    @Bean
-    @ConditionalOnExpression('${spinnaker.inMemory.enabled:false}')
-    ActionInstanceDao inMemoryActionInstanceDAO() {
-        new InMemoryActionInstanceDao()
-    }
+  @Autowired
+  TriggerListener triggerListener
 
-    @Bean
-    @ConditionalOnExpression('${spinnaker.inMemory.enabled:false}')
-    ExecutionDao inMemoryExecutionDao() {
-        new InMemoryExecutionDao()
+  @Bean
+  SchedulerFactoryBean schedulerFactoryBean(
+    Optional<DataSource> dataSourceOptional,
+    TriggerListener triggerListener,
+    JobDetail pipelineJobBean,
+    Trigger syncJobTrigger
+  ) {
+    SchedulerFactoryBean factoryBean = new SchedulerFactoryBean()
+    if (dataSourceOptional.present) {
+      factoryBean.dataSource = dataSourceOptional.get()
     }
 
-    @Bean
-    @ConditionalOnExpression('${spinnaker.inMemory.enabled:false}')
-    TriggerDao inMemoryTriggerDao() {
-        new InMemoryTriggerDao()
-    }
+    factoryBean.setGlobalTriggerListeners(triggerListener)
+    factoryBean.setJobDetails(pipelineJobBean)
+    factoryBean.setTriggers(syncJobTrigger)
 
-    @Bean
-    DaoConfigurer daoConfigurer(ActionInstanceDao actionInstanceDao, TriggerDao triggerDao, ExecutionDao executionDao) {
-        return new DaoConfigurer(actionInstanceDao, triggerDao, executionDao)
-    }
+    return factoryBean
+  }
 
-    @Bean
-    TriggerOperator triggerOperator(TriggerDao triggerDao,
-                                    @Value('${scheduler.threadPoolSize:10}') int threadPoolSize) {
-        new TriggerOperator(triggerDao, threadPoolSize)
-    }
+  /**
+   * Job for syncing pipeline triggers
+   */
+  @Bean
+  JobDetailFactoryBean pipelineSyncJobBean(
+    @Value('${scheduler.cron.timezone:America/Los_Angeles}') String timeZoneId
+  ) {
+    JobDetailFactoryBean syncJob = new JobDetailFactoryBean()
+    syncJob.jobClass = PipelineConfigsPollingJob.class
+    syncJob.jobDataMap.put("timeZoneId", timeZoneId)
+    syncJob.name = "Sync Pipelines"
+    syncJob.group = "Sync"
 
-    @Bean
-    ActionsOperator actionsOperator(TriggerOperator triggerOperator,
-                                    DaoConfigurer daoConfigurer,
-                                    @Value('${scheduler.threadPoolSize:10}') int threadPoolSize) {
-        new ActionsOperator(triggerOperator, daoConfigurer, threadPoolSize)
-    }
+    return syncJob
+  }
 
-    @Bean
-    ActionInstanceController actionInstanceController(ActionsOperator actionsOperator) {
-        new ActionInstanceController(actionsOperator: actionsOperator)
-    }
+  /**
+   * Trigger for the job to sync pipeline triggers
+   */
+  @Bean
+  @ConditionalOnExpression('${scheduler.enabled:false} && ${scheduler.pipelineConfigsPoller.enabled:true}')
+  SimpleTriggerFactoryBean syncJobTriggerBean(
+    @Value('${scheduler.pipelineConfigsPoller.pollingIntervalMs:30000}') long intervalMs,
+    @Autowired JobDetail pipelineSyncJobBean
```

Comment:
You don't need an autowired annotation here. That's implied by this being a bean method.

Reply:
thanks!